src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/SignatureHelp/SignatureInformation.cs

```diff
@@ -48,5 +63,42 @@ public override int GetHashCode()
         {
             return (Label, Documentation).GetHashCode();
         }
+
+        private static bool IsAIFunction(string label, out string name)
+        {
+            var names = new string[] { "AIClassify", "AIExtract", "AIModelPublish", "AIReply", "AISentiment", "AISummarize", "AISummarizeRecord", "AITranslate" };
+            foreach (var func in names)
+            {
+                if (label.Contains(func))
+                {
+                    name = func;
+                    return true;
+                }
+            }
+
+            name = null;
+            return false;
+        }
+    }
+
+    public class MarkdownDocumentation
+    {
+        public string Value { get; set; }
+    }
+
+    public class MarkdownSignatureInfomation
```

Comment:
>MarkdownSignatureInfomation [](http://example.com/codeflow?start=17&length=27)

Note this is missing some other things on SignatureInformation (like equals). Maybe some other pattern of class sharing?

Reply:
Yeah this was my "quick" solution but my idea was to have a base class called BaseSignatureInformation which would have only Label and Parameters field and Equals method that only compares label and then there would be child classes that would extend BaseSignatureInformation and add Documentation fields with two different types (string and MarkdownDocumentaiton) respectively. They both would overwrite the equals and gethashcode().

Reply:
Equals() and GetHashCode() are crucial to avoid duplicate signatures and this was the solution towards duplicate signatures. we need to make sure these methods are accurate with the planned changes to support markdown or it would case regression.